engines/director/lingo/lingo-preprocessor.cpp

```diff
@@ -191,6 +192,19 @@ Common::String Lingo::codePreprocessor(const char *s, ScriptType type, uint16 id
 			res1 = preprocessSound(res1);
 		}
 
+		if (type == kMovieScript && _vm->getVersion() <= 3 && !defFound) {
```

Comment:
This should be the first thing to do, right after the line is acquired, e.g. move it right after the `debugC`. Otherwise, you could potentially do unnecessary work with other preprocessing steps.

Reply:
Fixed in #2306